torch/csrc/autograd/engine.cpp

```diff
@@ -82,6 +82,7 @@ struct CompareFunctionTaskTime {
 struct ReadyQueue {
   std::priority_queue<FunctionTask, std::vector<FunctionTask>, CompareFunctionTaskTime> heap_;
   std::condition_variable not_empty_;
+  // For not_empty and to protect access to heap_
```

Comment:
I guess, to say a broader truth here: this protects all of the fields of this struct? (So, e.g., any read-write access to a ReadyQueue must take out this lock first? Is this true today?)